ospf6d/ospf6_top.c

```diff
@@ -1672,6 +1698,387 @@ DEFUN(show_ipv6_ospf6_route_type_detail, show_ipv6_ospf6_route_type_detail_cmd,
+       NO_STR
+       "External summary address\n"
+       "Specify IPv6 prefix\n"
+       "Adverise summary route to the AS \n")
+{
+	VTY_DECLVAR_CONTEXT(ospf6, ospf6);
+
+	struct prefix p;
+	int ret = CMD_SUCCESS;
+
+	ret = str2prefix(prefix_str, &p);
+	if (ret == 0) {
+		vty_out(vty, "Malformed prefix\n");
+		return CMD_WARNING_CONFIG_FAILED;
+	}
+
+	/* Apply mask for given prefix. */
+	apply_mask((struct prefix *)&p);
+
+	if (!ospf6_is_valid_summary_addr(vty, &p))
+		return CMD_WARNING_CONFIG_FAILED;
+
+	ret = ospf6_asbr_external_rt_advertise(ospf6, &p);
+	if (ret == OSPF6_INVALID)
+		vty_out(vty, "!!Invalid configuration\n");
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (ospf6_route_aggregation_timer,
+       ospf6_route_aggregation_timer_cmd,
+       "aggregation timer (5-1800)",
+       "External route aggregation\n"
+       "Delay timer (in seconds)\n"
+       "Timer interval(in seconds)\n")
+{
+	VTY_DECLVAR_CONTEXT(ospf6, ospf6);
+
+	ospf6_external_aggr_delay_timer_set(ospf6, timer);
+
+	return CMD_SUCCESS;
+}
+
+DEFPY (no_ospf6_route_aggregation_timer,
+       no_ospf6_route_aggregation_timer_cmd,
+       "no aggregation timer [5-1800]",
+       NO_STR
+       "External route aggregation\n"
+       "Delay timer\n"
+       "Timer interval(in seconds)\n")
+{
+	VTY_DECLVAR_CONTEXT(ospf6, ospf6);
+
+	ospf6_external_aggr_delay_timer_set(ospf6,
+			OSPF6_EXTL_AGGR_DEFAULT_DELAY);
+	return CMD_SUCCESS;
+}
+
+static int
+ospf6_print_vty_external_routes_walkcb(struct hash_bucket *bucket, void *arg)
+{
+	struct ospf6_route *rt = bucket->data;
+	struct vty *vty = (struct vty *)arg;
+	static unsigned int count;
+
+	vty_out(vty, "%pFX ", &rt->prefix);
+
+	count++;
+
+	if (count%5 == 0)
+		vty_out(vty, "\n");
+
+	if (OSPF6_EXTERNAL_RT_COUNT(rt->aggr_route) == count)
+		count = 0;
+
+	return HASHWALK_CONTINUE;
+}
+
+static int
+ospf6_print_json_external_routes_walkcb(struct hash_bucket *bucket, void *arg)
+{
+	struct ospf6_route *rt = bucket->data;
+	struct json_object *json = (struct json_object *)arg;
+	char buf[PREFIX2STR_BUFFER];
+	char exnalbuf[20];
+	static unsigned int count;
+
+	prefix2str(&rt->prefix, buf, sizeof(buf));
+
+	snprintf(exnalbuf, 20, "Exnl Addr-%d", count);
+
+	json_object_string_add(json, exnalbuf, buf);
+
+	count++;
+
+	if (OSPF6_EXTERNAL_RT_COUNT(rt->aggr_route) == count)
+		count = 0;
+
+	return HASHWALK_CONTINUE;
+}
+
+static int
+ospf6_show_summary_address(struct vty *vty, struct ospf6 *ospf6,
+			json_object *json,
+			bool uj, const char *detail)
+{
+	struct route_node *rn;
+	static const char header[] = "Summary-address       Metric-type     Metric     Tag         External_Rt_count\n";
+
+	if (!uj) {
+		vty_out(vty, "aggregation delay interval :%d(in seconds)\n\n",
+				ospf6->aggr_delay_interval);
+		vty_out(vty, "%s\n", header);
+	} else {
+		json_object_int_add(json, "aggregation delay interval",
+				ospf6->aggr_delay_interval);
+	}
+
+	for (rn = route_top(ospf6->rt_aggr_tbl); rn; rn = route_next(rn))
+		if (rn->info) {
```

Comment:
Please remove this check if it is not possible to happen.

There is a lot of indentation that can be removed in this function.